src/lib/datatypes/include/sol-memdesc.h

```diff
@@ -1085,6 +1086,525 @@ sol_memdesc_get_structure_member_memory(const struct sol_memdesc *structure_desc
+        return *(const long *)memory;
+    case SOL_MEMDESC_TYPE_SSIZE:
+        return *(const ssize_t *)memory;
+    case SOL_MEMDESC_TYPE_BOOLEAN:
+        return *(const bool *)memory;
+    case SOL_MEMDESC_TYPE_DOUBLE:
+        return *(const double *)memory;
+    case SOL_MEMDESC_TYPE_STRING:
+    case SOL_MEMDESC_TYPE_CONST_STRING:
+    case SOL_MEMDESC_TYPE_PTR:
+    case SOL_MEMDESC_TYPE_STRUCTURE:
+    case SOL_MEMDESC_TYPE_ARRAY:
+    default:
+        errno = EINVAL;
+        return 0;
+    }
+
+check_overflow:
+    if (u64 > INT64_MAX) {
+        errno = ERANGE;
+        return 0;
+    }
+    return u64;
+}
+
+/**
+ * @brief Helper to check if type is unsigned integer-compatible.
+ *
+ * @param desc the memory description.
+ *
+ * @return true if it is unsigned integer (uint8_t, uint16_t, uint32_t, uint64_t,
+ *         unsigned long or size_t).
+ */
+static inline bool
+sol_memdesc_is_unsigned(const struct sol_memdesc *desc)
+{
+    errno = EINVAL;
+    if (!desc)
+        return false;
+
+#ifndef SOL_NO_API_VERSION
+    if (desc->api_version != SOL_MEMDESC_API_VERSION_COMPILED)
+        return false;
+#endif
+
+    errno = 0;
+    switch (desc->type) {
+    case SOL_MEMDESC_TYPE_UINT8:
+    case SOL_MEMDESC_TYPE_UINT16:
+    case SOL_MEMDESC_TYPE_UINT32:
+    case SOL_MEMDESC_TYPE_UINT64:
+    case SOL_MEMDESC_TYPE_ULONG:
+    case SOL_MEMDESC_TYPE_SIZE:
+        return true;
+    case SOL_MEMDESC_TYPE_INT8:
+    case SOL_MEMDESC_TYPE_INT16:
+    case SOL_MEMDESC_TYPE_INT32:
+    case SOL_MEMDESC_TYPE_INT64:
+    case SOL_MEMDESC_TYPE_LONG:
+    case SOL_MEMDESC_TYPE_SSIZE:
+    case SOL_MEMDESC_TYPE_BOOLEAN:
+    case SOL_MEMDESC_TYPE_DOUBLE:
+    case SOL_MEMDESC_TYPE_STRING:
+    case SOL_MEMDESC_TYPE_CONST_STRING:
+    case SOL_MEMDESC_TYPE_PTR:
+    case SOL_MEMDESC_TYPE_STRUCTURE:
+    case SOL_MEMDESC_TYPE_ARRAY:
+    default:
+        return false;
+    }
+}
+
+/**
+ * @brief Helper to check if type is signed integer-compatible.
+ *
+ * @param desc the memory description.
+ *
+ * @return true if it is signed integer (int8_t, int16_t, int32_t, int64_t,
+ *         long or ssize_t).
+ */
+static inline bool
+sol_memdesc_is_signed(const struct sol_memdesc *desc)
+{
+    errno = EINVAL;
+    if (!desc)
+        return false;
+
+#ifndef SOL_NO_API_VERSION
+    if (desc->api_version != SOL_MEMDESC_API_VERSION_COMPILED)
+        return false;
+#endif
+
+    errno = 0;
+    switch (desc->type) {
+    case SOL_MEMDESC_TYPE_UINT8:
+    case SOL_MEMDESC_TYPE_UINT16:
+    case SOL_MEMDESC_TYPE_UINT32:
+    case SOL_MEMDESC_TYPE_UINT64:
+    case SOL_MEMDESC_TYPE_ULONG:
+    case SOL_MEMDESC_TYPE_SIZE:
+        return false;
+    case SOL_MEMDESC_TYPE_INT8:
+    case SOL_MEMDESC_TYPE_INT16:
+    case SOL_MEMDESC_TYPE_INT32:
+    case SOL_MEMDESC_TYPE_INT64:
+    case SOL_MEMDESC_TYPE_LONG:
+    case SOL_MEMDESC_TYPE_SSIZE:
+        return true;
+    case SOL_MEMDESC_TYPE_BOOLEAN:
+    case SOL_MEMDESC_TYPE_DOUBLE:
```

Comment:
Err... now I saw documentation... why not used `is_signed_integer` in the name of this function?